junit-platform-engine/src/main/java/org/junit/platform/engine/support/store/NamespacedHierarchicalStore.java

```diff
@@ -252,7 +252,8 @@ private <T> T castToRequiredType(Object key, Object value, Class<T> requiredType
 		}
 		// else
 		throw new NamespacedHierarchicalStoreException(
-			String.format("Object stored under key [%s] is not of required type [%s]", key, requiredType.getName()));
+			String.format("Object stored under key [%s] is not of required type [%s], but was [%s]: %s", key,
+				requiredType.getName(), value.getClass(), value));
```

Comment:
For future reference, please note that we typically invoke `Class#getName` when including types in messages, analogous to `requireType.getName()` on the same line.

I'll polish that on `main` since this has already been merged.

Thanks for the PR! 👍

Reply:
Makes sense, I overlooked that it included the `class ` prefix.